### PR TITLE
Install script

### PR DESCRIPTION
**Pull request by pengnam**
**User Story**
As a self-respecting user using any self-respecting project, user expects an install script to install a script

**TODO**
Hopefully someday will modify to work on windows

image

### CODE REVIEW

**Comment by juxd** [comment]
yo i think we need an install script for this isntall script

File: README [hunk 1]
```diff
@@ -65,8 +65,7 @@ EXAMPLES
     ./socprint.sh list d-lee
 
   To download and run from any directory:
-    sudo curl https://raw.githubusercontent.com/dlqs/SOCprint/master/socprint.sh -o /usr/local/bin/socprint.sh
-    sudo chmod 755 /usr/local/bin/socprint.sh
+    sh -c "$(curl -fsSL https://raw.githubusercontent.com/dlqs/SOCprint/master/install.sh)"
```
Comment (afterdusk) [inline README]: Is there no need for sudo?

Comment (pengnam) [reply README]: You right

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+
+# Credits: Modified oh-my-zsh install script
+# 1. Handle windows
+# 2.
+
+COMMAND="socprint"
```
Comment (afterdusk) [inline install.sh:4]: Should this be `socprint.sh` instead?

Comment (pengnam) [reply install.sh:4]: I prefer `socprint`

File: README [hunk 3]
```diff
@@ -65,8 +65,7 @@ EXAMPLES
     ./socprint.sh list d-lee
 
   To download and run from any directory:
-    sudo curl https://raw.githubusercontent.com/dlqs/SOCprint/master/socprint.sh -o /usr/local/bin/socprint.sh
-    sudo chmod 755 /usr/local/bin/socprint.sh
+    sudo sh -c "$(curl -fsSL https://raw.githubusercontent.com/dlqs/SOCprint/master/install.sh)"
```
Comment (juxd) [inline README:68]:
```suggestion
    curl -fsSL https://raw.githubusercontent.com/dlqs/SOCprint/master/install.sh | sudo sh 
```

File: install.sh (hunk 2 above)
Comment (afterdusk) [reply install.sh:4]: This breaks the update routine, which assumes that [the current script is installed as `socprint.sh`](https://github.com/dlqs/SOCprint/blob/c01d27b10e4cacfe90f19b03d5d85c50eed59158/socprint.sh#L266)

Also the update message instructs users to [install the new script as `socprint.sh`](https://github.com/dlqs/SOCprint/blob/c01d27b10e4cacfe90f19b03d5d85c50eed59158/socprint.sh#L138)

**Comment by dlqs** [comment]
Might need to iterate on this; thinking about ways to display a welcome banner without downloading an entire new script / is it possible to inline this?

File: install.sh (hunk 2 above)
Comment (pengnam) [reply install.sh:4]: @dlqs I just renamed all `socprint.sh` to `socprint`. Thoughts?

**Comment by pengnam** [comment]
> Might need to iterate on this; thinking about ways to display a welcome banner without downloading an entire new script / is it possible to inline this?

I guess since we actually curl and evaluate immediately its not saved in a file and its sorta inline

**Comment by pengnam** [comment]
@dlqs So should we merge or close this? Lmk I'm fine either way. I think its cute and its one of those nice features that makes a product more enjoyable. If you are worried about too many scripts I can start a `tools` folder and put this install script in it.

**Comment by juxd** [comment]
let's close this issue for now? we can reopen it when we have a better idea on where this PR should go.

**Pull request closed without merging**